### ISSUE TITLE
recorder.text(process_text) does not stop recording

### ISSUE THREAD

**Issue by SwamiKannan** [KoljaB/RealtimeSTT#79]
I am running the test file at [realtimestt_test.py](https://github.com/KoljaB/RealtimeSTT/blob/master/tests/realtimestt_test.py). The speech to text is working decently but the while loop does not stop when there is silence (or ambient noise)

```
while True:
        recorder.text(process_text)
```

Is there a way to set the library to threshold the ambient noise ? No new text is generated but the loop does not break. Hence, I cannot use the text generated for downstream applications

**Comment by KoljaB** [comment]
There's currently no noise reduction implemented which would offer such a threshold. While there are libraries that can do this in realtime like https://github.com/Rikorose/DeepFilterNet. But they are hard to install (in this case it starts with "Install cargo via [rustup](https://rustup.rs/)"). 

There are silero_sensitivity, webrtc_sensitivity and wake_words_sensitivity to take care for ambient noises. 

The loop isn't supposed to break currently. I did not really get your usecase scenario, can you describe how you want the library to behave? Maybe I can find a solution then.

**Comment by SwamiKannan** [comment]
Thanks a lot ! 
So I am using this library to convert to speech '**s**' to text '**t**' in real-time  and then post-conversion of the entire speech '**s**', pass the text '**t'** as an argument to a downstream function for processing (via an LLM user query). This is the first time I am using this library so this may be a very basic question. Thanks for your patience...

**Comment by KoljaB** [comment]
I'm asking specifically because of this: "No new text is generated but the loop does not break."
In which cases would you want to exit the loop? What would be your expected behaviour?

**Comment by SwamiKannan** [comment]
Sure, my understanding of the while loop was as follows:
```
While there is speech still to be transcribed to text:
             keep processing the speech
```
Hence, my understanding was that once the speech is completely transcribed, the while loop would break and the complete text can be extracted via the ```displayed_text``` variable. In this context, what I meant was when there is ONLY ambient noise, no new / junk text is generated. Which is **perfect** and indicates that the library understands that there is no new voice samples. However, if there are no new voice samples, the while loop should break. Else two things happen:
1. I am unable to extract the transcribed text cos there is no indication (code-wise or some return value of the transcribed text) that the transcription is complete. 
2. If I want to run a new live- speech sample, it keeps adding to the existing ```displayed_text``` rather than start a new string. 

Long story short, I would like to extract the text once the transcription of whatever speech I had provided is complete. And then reset ```displayed_text``` to '' for the next sample.

Hopefully, this explains where I am stuck. Thanks!

**Comment by KoljaB** [comment]
recorder.text(process_text) is a blocking call that runs forever if there is no speech detected. If speech is detected it will return control after speech was finished (by detecting when voice activity ends). 
"Hence, my understanding was that once the speech is completely transcribed, the while loop would break and the complete text can be extracted via the displayed_text variable."
The while loop does never break to ensure permament listening. Of course you can just one single time call recorder.text without the while loop.
"In this context, what I meant was when there is ONLY ambient noise, no new / junk text is generated."
recorder.text should ne return junk text on ambient noises. If it does the voice activity detection failed and maybe needs to be recustomized to not be that sensitive.
"I am unable to extract the transcribed text cos there is no indication (code-wise or some return value of the transcribed text) that the transcription is complete."
When the transcription is complete, the callback process_text is called. So this should be your indication.
"If I want to run a new live- speech sample, it keeps adding to the existing displayed_text rather than start a new string."
This is due to the implementation in this "realtimestt_test.py" example code. There might be a misunderstanding about what "speech is complete" means. Every time the process_text is called, RealtimeSTT considers the speech as complete. So the example just adds all texts up which were delivered after a "speech complete" event. So you'd need to reset displayed_text to '' in your implementation of process_text.

**Comment by SwamiKannan** [comment]
Ok, thanks a lot. This makes a lot of sense !!!!
